File: opensource/data/src/main/java/com/sos/scheduler/engine/data/folder/JobChainPath.java
package com.sos.scheduler.engine.data.folder;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.sos.scheduler.engine.data.order.OrderId;
import com.sos.scheduler.engine.data.order.OrderKey;

public final class JobChainPath extends TypedPath {
    public JobChainPath(@JsonProperty("absolutePath") AbsolutePath path) {
        super(FileBasedType.jobChain, path);
    }

    public OrderKey orderKey(OrderId o) {
        return new OrderKey(this, o);
    }

    @JsonCreator
    public static JobChainPath of(String absolutePath) {
        return new JobChainPath(new AbsolutePath(absolutePath));
    }
}

File: opensource/data/src/main/scala/com/sos/scheduler/engine/data/folder/JobChainPath.scala
package com.sos.scheduler.engine.data.folder

import com.fasterxml.jackson.annotation.JsonCreator
import com.sos.scheduler.engine.data.order.OrderId
import com.sos.scheduler.engine.data.order.OrderKey

final case class JobChainPath(string: String)
extends TypedPath {

  assertIsAbsolute()

  def typ =
    FileBasedType.jobChain

  def orderKey(o: String): OrderKey =
    orderKey(new OrderId(o))

  def orderKey(o: OrderId): OrderKey =
    new OrderKey(this, o)
}


object JobChainPath {
  @JsonCreator def of(absolutePath: String): JobChainPath =
    new JobChainPath(absolutePath)

  def makeAbsolute(path: String) =
    new JobChainPath(AbsolutePath.makeAbsolute(path))
}
